### ISSUE TITLE
useSanityNextImage: TypeScript error "No overload matches this call"

### ISSUE THREAD

**Issue by aaronbentley** [lorenzodejong/next-sanity-image#53]
Hey, thank you for your work on this library, it makes working with Sanity assets in Next.js a breeze 😃

I'm trying building an image component, but I'm running into a TypeScript error that I'm not sure how to resolve (I'm still learning TypeScript so please bear with me).

TypeScript is yelling at me, with the error message of:

```
No overload matches this call.

Overload 1 of 3, '(sanityClient: SanityClientLike, image: null, options?: UseNextSanityImageOptions | undefined): null', gave the following error.
    Argument of type 'SanityClient' is not assignable to parameter of type 'SanityClientLike'.
      Property 'clientConfig' is missing in type 'SanityClient' but required in type 'SanityClientLike'.

Overload 2 of 3, '(sanityClient: SanityClientLike, image: SanityImageSource, options?: UseNextSanityImageOptions | undefined): UseNextSanityImageProps', gave the following error.
    Argument of type 'SanityClient' is not assignable to parameter of type 'SanityClientLike'.

Overload 3 of 3, '(sanityClient: SanityClientLike, image: SanityImageSource | null, options?: UseNextSanityImageOptions | undefined): UseNextSanityImageProps | null', gave the following error.
    Argument of type 'SanityClient' is not assignable to parameter of type 'SanityClientLike'.
```

Here's my code so far:

```ts
'use client'
import {
    ImageUrlBuilder,
    useNextSanityImage,
    UseNextSanityImageBuilderOptions,
    UseNextSanityImageProps
} from 'next-sanity-image'
import { default as NextImage } from 'next/image'
import { sanityClient } from '../../lib/sanity.client'

type ImageProps = {
    image: {
        asset: {
            altText?: string
            metadata: {
                lqip: string
                palette: {
                    dominant: {
                        background: string
                    }
                }
            }
        }
        [key: string]: any
    }
    priority?: boolean
    [key: string]: any
}

/**
 * Image component, composes NextImage
 */
const Image = ({ image, priority = false, ...rest }: ImageProps) => {
    /**
     * Get Sanity image props, allowing for custom imageBuilder transforms
     */
    const imageProps: UseNextSanityImageProps = useNextSanityImage(
        sanityClient,  // ⛔️ <- Here's where TypeScript yells at me!
        image,
        {
            imageBuilder: (
                imageUrlBuilder: ImageUrlBuilder,
                options: UseNextSanityImageBuilderOptions
            ) => {
                /**
                 * Destructure options
                 */
                const {
                    width = undefined,
                    // originalImageDimensions = {},
                    // croppedImageDimensions = {},
                    quality = undefined
                } = options

                return imageUrlBuilder
                    .width(
                        width ||
                            Math.min(
                                options.originalImageDimensions.width,
                                1920
                            )
                    )
                    .quality(quality || 75)
                    .fit('clip')
            }
        }
    )

    /**
     * If no image/asset props return null
     */
    if (!imageProps) return null

    /**
     * Destructure asset props
     */
    const {
        asset: {
            altText = '',
            metadata: {
                lqip = '',
                palette: {
                    dominant: { background = '' }
                }
            }
        }
    } = image

    return (
        <div className='relative block h-[50vh] w-screen'>
            <NextImage
                src={imageProps.src}
                loader={imageProps.loader}
                alt={altText}
                priority={priority}
                fill
                sizes='(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw'
                placeholder='blur'
                blurDataURL={lqip}
                quality={90}
                style={{
                    objectFit: 'cover',
                    backgroundColor: background
                }}
                {...rest}
            />
        </div>
    )
}

export default Image

```

And here's the file I define the Sanity Client config:

```ts
import { createClient } from 'next-sanity'

export const projectId = process.env.NEXT_PUBLIC_SANITY_PROJECT_ID!
export const dataset = process.env.NEXT_PUBLIC_SANITY_DATASET!
export const apiVersion = process.env.NEXT_PUBLIC_SANITY_API_VERSION!

export const sanityClient = createClient({
    projectId,
    dataset,
    apiVersion,
    useCdn: false
})
```

Whereabouts am I going wrong here? Have I missed something kinda silly? I'm assuming it's something I've missed or typed incorrectly?

I've only been working with TypeScript for a few weeks, here's hoping someone can educate me & point me in the right direction 🤞

**Comment by lorenzodejong** [comment]
It seems like you're using `next-sanity` instead of the supported `@sanity/client`. I haven't heard of the `next-sanity` library before, but apparently this exposes a `SanityClient` interface which isn't compatible with the expected types.

Check out one of the usage examples: https://github.com/lorenzodejong/next-sanity-image#usage.

**Comment by aaronbentley** [comment]
Thank you for the swift reply @lorenzodejong 🙌

I've updated my code to use the `@sanity/client` as recommended, however I found that as of [@sanity/client v5.0.0](https://github.com/sanity-io/client/releases/tag/v5.0.0) the `SanityClient` default export has been deprecated and replaced with the named export `createClient`.

<img width="1212" alt="Screenshot 2023-02-26 at 15 25 03" src="https://user-images.githubusercontent.com/3936200/221420058-57a12f40-0587-4f3b-b520-b0597703ff37.png">

Using the new `createClient` does work when running the project on my local dev server, it's just making TypeScript unhappy.

**Comment by lorenzodejong** [comment]
Thanks for letting me know. It seems like for now this library only supports 3.x. I'll try to create a release updating the version somewhere this week. If you want, you could also provide a PR for this if you want to speed up the process :).

**Comment by aaronbentley** [comment]
Thanks @lorenzodejong, I've submitted a [PR](https://github.com/lorenzodejong/next-sanity-image/pull/54) - I'm still learning the ropes with TypeScript so forgive me if it's not up to scratch.

**Comment by VictorH04** [comment]
Also experiencing this issue while using `next-sanity`. Happens with `@sanity/client` also.

**Comment by lorenzodejong** [comment]
I've just released version 6.0.0 which includes the PR of @aaronbentley, thanks for the effort!

If you run into new issues, please feel free to open a new ticket.
